### ISSUE TITLE
Access draw list for window+foreground at same time?

### ISSUE THREAD

**Issue by dbr** [imgui-rs/imgui-rs#488]
Is it required that only one `DrawListMut` exists, or could one DrawListMut exist for each of the draw list types (window, foreground, background)?

For example, I'd like to do something roughly like this:

```rust
fn example(ui: &imgui::Ui) {
    let wd = ui.get_window_draw_list();
    let fd = ui.get_foreground_draw_list();

    for 0..100 {
        wd.add_line(...);
        fg.add_line(...);
    }
}
```

The use case is I have a custom drawn widget thing, and want to be able to highlight it in certain circumstances by drawing it above *everything* (so the `channel_split` doesn't help in this case)

I can restructure things so the two drawing methods are separate, but it would be much neater if I could create the FG list while still keeping the reference to the window draw list

**Comment by thomcc** [comment]
Hmm, I'll have to look and see if there's anything actually dangerous that could happen due to that. I think it's probably fine.

**Comment by vinterskog** [comment]
I forked imgui-rs some time ago for exactly this (and only this) reason. I needed to have a handle to both window and background drawlist at the same time. That works fine of course. In fact, there is no such restriction in the C-version. It is certainly good practice to allow only one reference per draw list, but the limitation to only be able to use one draw list at a time is neither necessary nor helpful (to say the least).
